### PR TITLE
pica_state: Make use of std::array where applicable

### PR DESCRIPTION
**Pull request by lioncash**
Prevents any implicit array to pointer decay and also makes the interface of `WriteUniformFloatReg` always take a correctly sized array.

While we're at it we can make `Zero()` a little safer to use by checking if the to-be-zeroed type is trivially copyable. We can also resolve undefined behavior related to type punning in `WriteUniformFloatReg` as well.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5241)

### CODE REVIEW

File: src/video_core/pica.cpp [hunk 1]
```diff
@@ -59,10 +62,10 @@ void State::Reset() {
     Zero(immediate);
     primitive_assembler.Reconfigure(PipelineRegs::TriangleTopology::List);
     vs_float_regs_counter = 0;
-    Zero(vs_uniform_write_buffer);
+    vs_uniform_write_buffer.fill(0);
```
Comment (B3n30) [inline src/video_core/pica.cpp:65]: just for performance reasons: is fill as fast as memset. Theoretical it should be the same, but just to be sure, is there a way to check that?

Comment (lioncash) [reply src/video_core/pica.cpp:65]: Yes, it's the [same](https://godbolt.org/z/qaAnTt). Even if it wasn't, for arrays of 3 or 4 elements with primitives in them, the difference would be negligible.

**Comment by lioncash** [comment]
@Subv I've amended it in a way that should preserve backwards compatibility.

**Comment by B3n30** [comment]
please fix clang-format so I can merge this

**Comment by lioncash** [comment]
@B3n30 done

**Pull request merged**